en/ros2/px4_ros2_msg_translation_node.md

```diff
@@ -0,0 +1,434 @@
+# PX4 ROS 2 Message Translation Node
+
+<Badge type="tip" text="main (PX4 v1.16+)" /> <Badge type="warning" text="Experimental" />
+
+The message translation node allows ROS 2 applications that were compiled against different versions of the PX4 messages to interwork with newer versions of PX4, and vice versa, without having to change either the application or the PX4 side.
+
+## Overview
+
+The translation of messages from one definition version to another is possible thanks to the introduction of [message versioning](../middleware/uorb.md#message-versioning).
+
+The translation node has access to all message versions previously defined by PX4. It dynamically observes the DDS data space, monitoring the publications, subscriptions and services originating from either PX4 via the [uXRCE-DDS Bridge](../middleware/uxrce_dds.md), or ROS 2 applications. When necessary, it converts messages to the current versions expected by both applications and PX4, ensuring compatibility.
+
+![Overview ROS 2 Message Translation Node](../../assets/middleware/ros2/px4_ros2_interface_lib/translation_node.svg)
```

Comment:
I understand this diagram, but it took me a little while to understand what it meant - specifically what versions are supported. Do you think this makes it easier?

![image](https://github.com/user-attachments/assets/770cea06-5387-4471-824f-84c62bb7310a)

Reply:
Yes, that's the only change. 

Initially I was thinking of marking up the nodes with the versions of messages that they sub/pub because I wanted to be able to show what the translation node does and why. With colour coding I think we can avoid this.